Core/KSPManager.cs

```diff
@@ -200,67 +206,91 @@ public void CloneInstance (KSP existing_instance, string new_name, string new_pa
+                    fileMgr.CreateDirectory(Path.Combine(newPath, "Ships"));
+                    fileMgr.CreateDirectory(Path.Combine(newPath, "Ships", "VAB"));
+                    fileMgr.CreateDirectory(Path.Combine(newPath, "Ships", "SPH"));
+                    fileMgr.CreateDirectory(Path.Combine(newPath, "Ships", "@thumbs"));
+                    fileMgr.CreateDirectory(Path.Combine(newPath, "Ships", "@thumbs", "VAB"));
+                    fileMgr.CreateDirectory(Path.Combine(newPath, "Ships", "@thumbs", "SPH"));
+
+                    // Don't write the buildID.txts if we have no build, otherwise it would be -1.
+                    if (version.IsBuildDefined)
+                    {
+                        fileMgr.WriteAllText(Path.Combine(newPath, "buildID.txt"), String.Format("build id = {0}", version.Build));
+                        fileMgr.WriteAllText(Path.Combine(newPath, "buildID64.txt"), String.Format("build id = {0}", version.Build));
+                    }
+
+                    // Create the readme.txt WITHOUT build number.
+                    fileMgr.WriteAllText(Path.Combine(newPath, "readme.txt"), String.Format("Version {0}", new KspVersion(version.Major, version.Minor, version.Patch).ToString()));
+
+                    // Create the needed folder structure and the readme.txt for DLCs that should be simulated.
+                    if (dlcs != null)
+                    {
+                        foreach (KeyValuePair<DLC.IDlcDetector, KspVersion> dlc in dlcs)
+                        {
+                            DLC.IDlcDetector dlcDetector = dlc.Key;
+                            KspVersion dlcVersion = dlc.Value;
+
+                            if (!dlcDetector.AllowedOnBaseVersion(version))
+                                throw new IncorrectKSPVersionKraken(
+                                    String.Format("KSP version {0} or above is needed for {1} DLC.",
+                                        dlcDetector.ReleaseGameVersion,
+                                        dlcDetector.IdentifierBaseName
+                                ));
+
+                            string dlcDir = Path.Combine(newPath, dlcDetector.InstallPath());
+                            fileMgr.CreateDirectory(dlcDir);
+                            fileMgr.WriteAllText(
+                                Path.Combine(dlcDir, "readme.txt"),
+                                String.Format("Version {0}", dlcVersion));
+                        }
+                    }
+
+                    // Add the new instance to the registry
+                    KSP new_instance = new KSP(newPath, newName, User, false);
+                    AddInstance(new_instance);
+                }
+                // Thrown by AddInstance() if created instance is not valid.
+                // Thrown f.e. if a write operation didn't complete for unknown reasons.
+                catch (NotKSPDirKraken kraken)
```

Comment:
Unless I'm missing something, we don't need to catch and re-throw here. That's for when you want to throw a different exception or take some action in addition. If we just remove the try/catch blocks, the effect should be the same.

Reply:
Yeah, I added that for clarity, to point out what call can throw which exception and why. Don't know if that's a 'don't' in C#.

Reply:
Oh, OK. Nevermind then!